### ISSUE TITLE
Tax on Shipping when only Non-Taxable items in Cart and Flatrate Shipping

### ISSUE THREAD

**Issue by AdminLinenStore** [magento/magento2#16221]
### issue
We sell both taxable and non-taxable items from NY location so NY based shipping destination is required a sales tax charged based on item. We set up shipping to display separately and taxable (by default because there is no cart based method to choose from) and we cannot add tax to shipping price. In a simple explanation, shipping should only be charged sales tax if that order has 1 or more taxable items. 

### Preconditions

1.  Magento 2.2.1/2.2.2
2.  PHP 7.0 
3. MySQL 5.7 

### Expected Result
when adding non-taxable items in cart and select NY shipping address, sales tax shouldn't be charged from flatrate shipping (in our case it's $4.99)

### Actual Result
Sales tax is charged based on shipping price (in our case it's about $0.44 - NY rate is 8.875%)

### Steps to Reproduce
- set up tax rule for NY retail customer
![tax-rule](https://user-images.githubusercontent.com/20774450/41558550-26749950-730f-11e8-8e56-308699cb1ac3.png)

- set up a flatrate shipping of $4.99 and setting as below 
![flatrate499](https://user-images.githubusercontent.com/20774450/41557455-b3a31d14-730b-11e8-9389-599e8292fab6.png)

- set up for tax setting is as below 
![screencapture-oasissnacks-admin-admin-system_config-edit-section-tax-key-b10c1ab1a2cf7533a16c05d788855ed382fab434a55732402b98fdda7de9357e-2018-06-18-15_08_16](https://user-images.githubusercontent.com/20774450/41557473-c66ca80c-730b-11e8-8696-a7b90960f9ae.png)

- set up items with non-taxable / none tax class ( to not charging sales tax)

- place order for those items (only non-taxable in cart) select NY as shipping destination

- select flat rate shipping $4.99 

- tax is calculated for shipping 

*** not really sure it is a bug but obviously there is not other options for this kind of sales tax charged based on cart content or so **** or maybe i am missing something

***note : we want to display item price and shipping price without sales tax included and sales tax should show separately by itself on calculated. 

Thank you,
Biggs

**Comment by ghost** [comment]
Hello @AdminLinenStoreyou need to create own tax class and set the product to this none class tax, When i tested it everything worked fine

**Comment by AdminLinenStore** [comment]
@engcom-backlog-nazar , Thank you for your reply. I still don't understand what do you meant by create own tax class? I have product tax class set for non-taxable product as "None" but what I am having an issue with is a shipping tax class is being calculated when only non-taxable item is in cart as Tax Class for Shipping can either be set for None or Taxable one way or another. In may case, shipping tax should be calculated based on cart content. 

case 1 : if 1 or more taxable product is in cart, flat rate shipping is taxable
case 2 : if all products in cart are non-taxable, flat rate shipping is not taxable

so basically shipping is taxable or not, it depends on product in cart.

If you can share your set up i would really appreciate

**Comment by ghost** [comment]
@AdminLinenStore   you have 2 default  tax class ->1 none -> 2 "Taxable goods" so you need to create your "disabale" class tax, then add product to this class. and in tax rules uncheck your "disable" class.

**Comment by AdminLinenStore** [comment]
That's exactly my set up has been

**Comment by ghost** [comment]
Hi @AdminLinenStore I'm not able to reproduce , can you confirm that you reproduce this on vanilla instance ?

**Comment by ghost** [comment]
@AdminLinenStore , we are closing this issue due to inactivity. If you'd like to update it, please reopen the issue.

**Comment by webusat** [comment]
Hi @engcom-backlog-nazar, I have the same issue.  

I need to tax the shipping only if there is a taxable item in the cart.  If there is no taxable item in the cart then no tax is added to the shipping cost.

In other words, if there is a taxable item in the cart, then tax the shipping charges.

**Comment by Rus0** [comment]
MM18MX

**Comment by magento-engcom-team** [comment]
Hi @Rus0. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by Rus0** [comment]
#MM18MX
